lib/Horde/SessionHandler/Storage/File.php

```diff
@@ -129,14 +132,22 @@ public function write($id, $session_data)
     }
 
     /**
+     * @param string $id
+     * @return bool
      */
-    public function destroy($id)
+    public function destroy($id): bool
     {
         $this->close();
 
-        $filename = $this->_params['path'] . '/' . self::PREFIX . $id;
+        if (preg_match('/^[A-Za-z0-9,-]{22,256}$/', $id) !== 1) {
+            return false;
+        }
 
-        return @unlink($filename);
+        $filename = realpath($this->_params['path'] . '/' . self::PREFIX . $id);
+        if ($filename && (strpos($filename, $this->_params["path"]) == 0)) {
```

Comment:
this is not really needed after we validated the id, but I guess it does not hurt to be extra safe

Reply:
Good catch, this wasn't supposed to stay in there